src/Test/DefaultTest.php

```diff
@@ -53,20 +52,45 @@ public function execute(array $sourceData, array $destinationData): bool
     }
 
     /**
-     * @param  array<string, numeric-string|array<string, numeric-string>>|numeric-string  $data
-     * @param  array<string>  $fields
+     * Recursively checks if an offset exists in a nested array.
+     *
+     * This function is designed to handle nested arrays and verify the existence of
+     * a specified offset (string key) within them. It performs recursive checks on
+     * nested arrays to ensure that the offset exists at each level.
+     *
+     * @param array<string, numeric-string|array<string, numeric-string>>|numeric-string $data
+     *     The data array to check for the offset in.
+     * @param array<string> $fields
+     *     An array of string keys representing the path to the desired offset.
      *
      * @return bool
+     *     Returns `true` if the specified offset exists in the nested array,
+     *     `false` otherwise.
+     *
+     * Used a combination of type hinting and explicit checks to assist PHPStan
+     * in understanding the types involved and resolving static analysis errors.
      */
     private function issetField(mixed $data, array $fields): bool
     {
+        // Check if $data is an array
+        if (!is_array($data)) {
+            return false;
+        }
+
+        // Get the next key to check
         $key = array_shift($fields);
+
+        // If there are no more keys to check, the offset exists
         if ($key === null) {
             return true;
         }
 
-        $test = isset($data[$key]);
+        // Check if the key exists in the data array
+        if (!array_key_exists($key, $data)) {
```

Comment:
There is a subtle difference between array_key_exists and isset. Is this what you want to change here?

Reply:
@hussainweb Thank you for your question on usage of array_key_exists() instead of isset(). Yes, I agree that there is a subtle difference between array_key_exists() and isset(). I am using array_key_exists() instead of isset() because I want to check if the key exists in the array, even if the value is null. I think this is a safer way to check for key existence, because the function issetField() would be designed to check for the existence of a key in a nested array, regardless of the value. Please clarify.